core/src/main/java/com/microsoft/applicationinsights/internal/perfcounter/PerformanceCounter.java

```diff
@@ -0,0 +1,37 @@
+/*
```

Comment:
I think it is better to define it as an abstract class so you'll be able to enforce restrictions like timeouts and such.